core/Cargo.toml

```diff
@@ -31,6 +31,10 @@ math = { package = "winter-math", version = "0.9", default-features = false }
 #miden-crypto = { version = "0.9", default-features = false }
 miden-crypto = { git = "https://github.com/0xPolygonMiden/crypto", branch = "next", default-features = false }
 miden-formatting = { version = "0.1", default-features = false }
+miette = { version = "7.1.0", git = "https://github.com/bitwalker/miette", branch = "no-std", default-features = false, features = [
+    "fancy-no-syscall",
+    "derive",
+] }
```

Comment:
It's only needed for [here](https://github.com/0xPolygonMiden/miden-vm/blob/ceca4f6bc3ac1d3a6244c95ab5816c573a16e905/miden/src/cli/data.rs#L422-L425), but this would be better answered by @bitwalker

Reply:
The `miette` crate is used for error handling (i.e. defining error types), and also exports the trait referenced by @plafer's link (`WrapErr`). That stuff is all re-exported from the `miden-assembly` crate though (in the `diagnostics` namespace), so a direct dependency on `miette` is only needed when `miden-assembly` isn't a dependency.

I believe the reason it is added to `core` here is that @plafer defined a new error type using the new diagnostics infrastructure based on `miette`, but `core` can't depend on `miden-assembly` itself.

Reply:
Removed the dependency `miette` dependency from `core`: [d5c14c6](https://github.com/0xPolygonMiden/miden-vm/pull/1349/commits/d5c14c61de3dabdb64a112d22417f79497bb3ab7)